crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2469,7 +2472,13 @@ impl<'db> TypeInferenceBuilder<'db> {
             (UnaryOp::Invert, Type::BooleanLiteral(bool)) => Type::IntLiteral(!i64::from(bool)),
 
             (UnaryOp::Not, ty) => ty.bool(self.db).negate().into_type(self.db),
-
+            (_, Type::Any) => Type::Any,
+            (_, Type::Unknown) => Type::Unknown,
+            (op, Type::Instance(class)) => {
+                let class_member = class.class_member(self.db, op.dunder());
+                let call = class_member.call(self.db, &[operand_type]);
```

Comment:
If we get a not-callable outcome here, we should emit an unsupported-operation diagnostic (in the draft PR https://github.com/astral-sh/ruff/pull/13800 we are currently proposing to use the code `operation` for that diagnostic, though I'm kinda thinking I might prefer the more explicit `unsupported-operation` -- @AlexWaygood ?).

If you'd rather not implement that in this PR, that's fine, but please add a TODO comment for it here.

Reply:
> in the draft PR https://github.com/astral-sh/ruff/pull/13800 we are currently proposing to use the code `operation` for that diagnostic, though I'm kinda thinking I might prefer the more explicit `unsupported-operation` -- @AlexWaygood ?

We actually use the `operator` code in #13800. I don't mind using the `unsupported-` prefix, but I would much prefer to use the operat-_or_ suffix rather than operat-_ion_. To some extent any type checker diagnostic comes about as a result of an "unsupported operation"; but this error code specifically deals with attempts to use an operat-_or_ in an unsupported way.

TL;DR: `unsupported-operator`?

Reply:
Oops, my mistake; yeah `unsupported-operator` sounds good to me!

Reply:
Thanks do we want to make functions to create the error with the right prefix in the future? I just put the string in the error message now.

Reply:
At some point we'll make our diagnostics less "stringly typed", and we'll probably do an audit of all our error codes at around the same time. For now I wouldn't much worry about :-)